airbyte-integrations/connectors/destination-langchain/destination_langchain/processor.py

```diff
@@ -0,0 +1,95 @@
+from typing import List, Optional, Tuple, Mapping
+from destination_langchain.config import ProcessingConfigModel
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from airbyte_cdk.models import AirbyteRecordMessage, ConfiguredAirbyteCatalog, ConfiguredAirbyteStream
+from langchain.document_loaders.base import Document
+from langchain.utils import stringify_dict
+import dpath.util
+from dpath.exceptions import PathNotFound
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+import uuid
+from dataclasses import dataclass
+
+METADATA_STREAM_FIELD = "_airbyte_stream"
+METADATA_NATURAL_ID_FIELD = "_natural_id"
+
+class Processor:
+    streams: Mapping[str, ConfiguredAirbyteStream]
+
+    def __init__(self, config: ProcessingConfigModel, catalog: ConfiguredAirbyteCatalog, max_metadata_size: Optional[int] = None):
+        self.streams = {stream.stream.name: stream for stream in catalog.streams}
```

Comment:
fyi the stream could have a namespace

Reply:
Switched to use namespace and stream name as identifier